src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs

```diff
@@ -0,0 +1,738 @@
+		}
+
+		/// <summary>
+		/// Returns true if the property has an accessible public setter and has a parameterless constructor
+		/// </summary>
+		private bool IsNewableProperty(IPropertySymbol property, out string newableTypeName)
+		{
+			var namedType = property.Type as INamedTypeSymbol;
+
+			var isNewable = property.SetMethod.SelectOrDefault(m => m.DeclaredAccessibility == Accessibility.Public, false) &&
+				namedType.SelectOrDefault(nts => nts.Constructors.Any(ms => ms.Parameters.Length == 0), false);
+
+			newableTypeName = isNewable ? GetFullGenericTypeName(namedType) : null;
+
+			return isNewable;
+		}
+
+		/// <summary>
+		/// Returns true if the type implements either ICollection, IList or one of their generics
+		/// </summary>
+		private bool IsCollectionOrListType(INamedTypeSymbol propertyType)
+			=> IsImplementingInterface(propertyType, _iCollectionSymbol)
+			|| IsImplementingInterface(propertyType, _iCollectionOfTSymbol)
+			|| IsImplementingInterface(propertyType, _iListSymbol)
+			|| IsImplementingInterface(propertyType, _iListOfTSymbol);
+
+		/// <summary>
+		/// Returns true if the type exactly implements either ICollection, IList or one of their generics
+		/// </summary>
+		private bool IsExactlyCollectionOrListType(INamedTypeSymbol type)
+		{
+			return type == _iCollectionSymbol
+				|| type.OriginalDefinition == _iCollectionOfTSymbol
+				|| type == _iListSymbol
+				|| type.OriginalDefinition == _iListOfTSymbol;
+		}
+
+		/// <summary>
+		/// Determines if the provided object definition is of a C# initializable list
+		/// </summary>
+		private bool IsInitializableCollection(XamlObjectDefinition definition)
+		{
+			if (definition.Members.Any(m => m.Member.Name != "_UnknownContent"))
+			{
+				return false;
+			}
+
+			var type = FindType(definition.Type);
+			if (type == null)
+			{
+				return false;
+			}
+
+			return IsImplementingInterface(type, _iCollectionSymbol)
+				|| IsImplementingInterface(type, _iCollectionOfTSymbol);
+		}
+
+		/// <summary>
+		/// Gets the 
+		/// </summary>
+		private IPropertySymbol GetPropertyByName(XamlType declaringType, string propertyName)
+		{
+			var type = FindType(declaringType);
+
+			return type?.GetAllProperties().FirstOrDefault(p => p.Name == propertyName);
+		}
+
+		private static bool IsDouble(string typeName)
+		{
+			// handles cases where name is "Java.Lang.Double"
+			// TODO: Fix type resolution
+			return typeName.Equals("double", StringComparison.InvariantCultureIgnoreCase)
+				|| typeName.EndsWith(".double", StringComparison.InvariantCultureIgnoreCase);
+		}
+
+		private bool IsString(XamlObjectDefinition xamlObjectDefinition)
+		{
+			return xamlObjectDefinition.Type.Name == "String";
+		}
+
+		private bool IsPrimitive(XamlObjectDefinition xamlObjectDefinition)
+		{
+			switch (xamlObjectDefinition.Type.Name)
+			{
+				case "Double":
+				case "Int32":
+				case "Int16":
+				case "Single":
+				case "Boolean":
```

Comment:
What about  `Int64`, `UIntXX` and `Byte` (and other value types like `enum` and `char` ?)